### ISSUE TITLE
User request: allow URL links for Google doc activity

### ISSUE THREAD

**Issue by oliver-phet** [phetsims/website#494]
Thank you for providing such a wonderful resource for science teachers!

I've created several resources to guide students through various lessons utilizing PhET simulations, however they are in Google Docs format.   I could, of course, convert these to Word formats to upload to the PhET site, however think it would be valuable if the ability to hyperlink to publicly accessible Google Doc would be beneficial.   Google Apps learning environments are increasingly ubiquitous in education (I know CU, itself, utilizes Google Apps), even more so in the sorts of one-to-one laptop/tablet learning environments where PhET similations are likely to be most utilized.   

Is there any possibility that the resource-upload portal could add a 'link-to' option, rather than document upload, particularly for Google Doc type resources?

**Comment by oliver-phet** [comment]
We've had a similar request before. And we decided not to pursue due to the risk of allowing users to upload any URL. Could it be possible to only allow URL's that are from a Google doc?

**Comment by mattpen** [comment]
Yes, we could filter based on the whether the URL starts with "https://docs.google.com", although there may be a possibility of cross-site-scripting.

**Comment by ariel-phet** [comment]
Well, we want them in word format (ideally) from the teacher. Many other users that do not use google docs will appreciate the activity in Word. 

Can we explain this to the user and also let them know they are certainly welcome to put the link in the word doc (something like..."I created this activity in google docs, to see that document directly go here"

**Comment by ariel-phet** [comment]
I think if we did allow this functionality we would need to vet people...just as we mark "trusted translators" we would need to view activities and verify the link is good, then mark "trusted contributor" or such

**Comment by amanda-phet** [comment]
I agree with Ariel that they need to have a word version, but if they want to also share the google doc link somewhere we should make that a possibility. 

>I think if we did allow this functionality we would need to vet people 
How would we do this? How could we verify the link is good without clicking it ourselves?

**Comment by oliver-phet** [comment]
> Well, we want them in word format (ideally) from the teacher. Many other users that do not use google docs will appreciate the activity in Word.

All of those Word-only users can still view and download a google doc as a word file. In fact, it might be a better experience than having to download a Word file.

> Can we explain this to the user and also let them know they are certainly welcome to put the link in the word doc (something like..."I created this activity in google docs, to see that document directly go here"

How would a user know there is a google doc version? Is this the best user experience for Chromebook and tablet users?

This would also help alleviate version issues. The google teacher can update their activity without having to remember to upload the new version to our site. And teachers using the activity will always have access to the latest version.

**Comment by oliver-phet** [comment]
> I think if we did allow this functionality we would need to vet people...just as we mark "trusted translators" we would need to view activities and verify the link is good, then mark "trusted contributor" or such

What is the current policy on viewing activity contributions? Couldn't a malicious user upload an insecure file on the site right now?

**Comment by ariel-phet** [comment]
Well, usually @amyh-phet looks over every contribution each week

**Comment by oliver-phet** [comment]
I'll mockup a new design of the submission page that includes a google doc URL. 
@mattpen Any idea if we could tell, based on the URL, if a google doc is set to 'View Only'? We'd like to ensure that users upload as 'View Only' setting to avoid others modifying the doc.
